### PR TITLE
Feature/wan support

### PR DESCRIPTION
**Pull request by smousa**
This code is used to ensure service connectivity during a WAN network outage.  Connection timeouts are set by pool and are configurable by CLI only. UI will be available in a separate PR.

Pool connection timeouts should be set based on the max difference in latency between the hosts and the master.

The minimum observed timeout is 2 x the configured connection timeout.  This is to allow a quiesce time when the network is perceived down, since not all hosts may necessarily show that they are down at the exact same time and the time to wait after connectivity has been restored to the network.

I am happy to explain the architecture to anyone who is interested in learning how the scheduling system works and how this change impacts scheduling as a whole.

### CODE REVIEW

**Comment by smousa** [comment]
retest this

File: zzk/service/hostregistry.go [hunk 1]
```diff
@@ -14,273 +14,449 @@
...
-
-var (
-	ErrHostInvalid = errors.New("invalid host")
-	ErrShutdown    = errors.New("listener shut down")
-)
-
-func hostregpath(nodes ...string) string {
-	p := append([]string{zkRegistry, zkHost}, nodes...)
-	return path.Clean(path.Join(p...))
+// HostRegistryListener is a monitors the availability of hosts within a pool
```
Comment (ed-) [inline zzk/service/hostregistry.go]: s/is a //

File: cli/cmd/pool.go [hunk 2]
```diff
@@ -319,3 +326,37 @@ func (c *ServicedCli) cmdRemoveVirtualIP(ctx *cli.Context) {
...
+// serviced pool set-conn-timeout POOLID TIMEOUT
+func (c *ServicedCli) cmdSetConnTimeout(ctx *cli.Context) {
+	args := ctx.Args()
+	if len(args) != 2 {
+		fmt.Printf("Incorrect Usage.\n\n")
+		cli.ShowCommandHelp(ctx, "set-conn-timeout")
+		return
+	}
+
+	connTimeout, err := time.ParseDuration(args[1])
```
Comment (rphillipsz) [inline cli/cmd/pool.go]: We should have some cli tests to verify that negative numbers are rejected, and that 0 or positive numbers are accepted as valid.

File: zzk/service/hoststate.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+func NewHostState(state *servicestate.ServiceState) *HostState {
+	return &HostState{
+		HostID:         state.HostID,
+		ServiceID:      state.ServiceID,
+		ServiceStateID: state.ID,
+		DesiredState:   int(service.SVCRun),
+	}
+}
+
+// Version inplements client.Node
```
Comment (ed-) [inline zzk/service/hoststate.go]: s/inp/imp/

File: cli/cmd/pool.go (hunk 2 above)
Comment (smousa) [inline cli/cmd/pool.go]: added that to the domain level

**Pull request merged**